### PR TITLE
solution 6.2

### PR DESCRIPTION
**Pull request by Dubovaya**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 42;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: why 42?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +22,32 @@ public Integer getLength() {
...
+            return true;
+        }
+        if (obj == null || !obj.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        Rectangle objectCast = (Rectangle) obj;
+        if (length == null && width == null && color == null && objectCast.color == null
+                && objectCast.width == null && objectCast.length == null) {
+            return hashCode() == objectCast.hashCode();
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: Do you need to complare hashcodes in equals?

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: let's add these checks below

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (Dubovaya) [reply src/main/java/core/basesyntax/Rectangle.java]: Got to now, change to 47 (prime).

**Comment by Dubovaya** [comment]
change my 42 to prime 47

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: what about this comment?

Comment (Dubovaya) [reply src/main/java/core/basesyntax/Rectangle.java]: oh, didn't see this before, too much tabs.

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**